src/firefly/java/edu/caltech/ipac/table/PrimitiveList.java

```diff
@@ -222,7 +241,7 @@ public Object get(int idx) {
         public void set(int idx, Object val) {
             checkType(val);
             ensureCapacity(idx);
-            data[idx] = val == null ? Integer.MIN_VALUE : (int) val;
+            data[idx] = val == null ? Integer.MIN_VALUE : getIntValue(val);
```

Comment:
You can cast Byte or Short into int.  Will just have to re-arrange the code slightly so that we only do ensureCapacity when `val` can be casted.. i.e.
```
    int ival = val == null ? Integer.MIN_VALUE : (int) val;
    ensureCapacity(idx);
    data[idx] = ival;
```
Sorry if you have to remove your code.